packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -70,53 +74,18 @@ export class CallContext {
     return this._state;
   }
 
-  public setState(state: CallClientState): void {
-    this._state = state;
+  public modifyState(modifier: (draft: CallClientState) => void): void {
+    this._state = produce(this._state, modifier, (patches: Patch[]) => {
+      if (getLogLevel() === 'verbose') {
+        // Log to `info` because AzureLogger.verbose() doesn't show up in console.
+        this._logger.info(`State change: ${JSON.stringify(patches)}`);
+      }
+    });
     if (!this._batchMode) {
```

Comment:
Wouldn't this always be false now that no other function is setting it?

Reply:
Good point. Will remove it in my follow up PR.

Reply:
Done in #1451